prac_08/convert_miles_km.py

```diff
@@ -0,0 +1,54 @@
+"""
+
+CP1404 Practical
+Kivy GUI program to convert miles to kilometres
+Joshua Delos Santos
+Started 05/11/2023
+
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+from kivy.properties import StringProperty
+
+MILE_TO_KILOMETRE_RATE = 1.60934
+
+
+class ConvertMilesToKilometresAPP(App):
+    """ConvertMilesToKilometresAPP is an app for converting miles to kilometres."""
+    message = StringProperty()
+
+    def build(self):
+        """Build the Kivy app from the kv file"""
+        Window.size = (500, 400)
+        self.title = "Convert Miles to Kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        self.message = "Type in the field & press convert"
+        return self.root
+
+    def handle_calculate(self):
+        """Handle calculation for converting miles to kilometres."""
+        try:
+            result = float(self.root.ids.input_number.text) * MILE_TO_KILOMETRE_RATE
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
```

Comment:
You can use a function instead of doing these ValueError exception checks multiple times 👍

Reply:
Good thinking SRP!